staging/src/k8s.io/legacy-cloud-providers/azure/azure_config.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	cloudConfigNamespace = "kube-system"
+	cloudConfigKey       = "cloud-config"
+)
+
+// The configure scope for Azure cloud provider secret. Supported values are:
+// * all            : configure applied for components (kubelet and controller-manager). This is the default value.
+// * node           : configure applied for nodes (kubelet).
+// * control-plane  : configure applied for control plane components (controller-manager).
+//
+// For different configure scope, the secret name would also be different:
+// * all            : secret name would be azure-cloud-provider.
+// * node           : secret name would azure-cloud-provider-node.
+// * control-plane  : secret name would be azure-cloud-provider-control-plane.
+type cloudConfigScope string
```

Comment:
I don't agree with these scope values, seems backwards to what we are trying to push for. Secrets should only be available for control plane, if we let clusters define secrets for nodes it make migration harder in the future.

Reply:
yep, would remove this in this PR.